test/Microsoft.OpenApi.Tests/Models/OpenApiOperationTests.cs

```diff
@@ -859,5 +860,34 @@ public void OpenApiOperationCopyConstructorWithAnnotationsSucceeds()
 
             Assert.NotEqual(baseOperation.Metadata["key1"], actualOperation.Metadata["key1"]);
         }
+
+        [Theory]
+        [InlineData(OpenApiSpecVersion.OpenApi2_0)]
+        [InlineData(OpenApiSpecVersion.OpenApi3_0)]
+        [InlineData(OpenApiSpecVersion.OpenApi3_1)]
+        public async Task SerializeAsJsonAsyncThrowsIfTagReferenceIsUnresolved(OpenApiSpecVersion version)
+        {
+            var document = new OpenApiDocument()
+            {
+                Tags =
+                [
+                    new() { Name = "one" },
+                    new() { Name = "three" }
+                ]
+            };
+
+            var operation = new OpenApiOperation()
+            {
+                Tags =
+                [
+                    new OpenApiTagReference("one", document),
+                    new OpenApiTagReference("two", document),
+                    new OpenApiTagReference("three", document)
+                ]
+            };
+
+            var exception = await Assert.ThrowsAsync<OpenApiException>(() => operation.SerializeAsJsonAsync(version));
+            Assert.Equal("The OpenAPI tag reference 'two' does reference a valid tag.", exception.Message);
```

Comment:
@martincostello Looks like this needs to be "does not".

Reply:
Oops, you're right. Do you want to do a PR to fix it as you found it?

Reply:
Not really :)

Reply:
Ok I'll fix it shortly then. Should take less than 5 minutes.

Reply:
#2345